### PR TITLE
Fix bug in stage_mem

### PR DESCRIPTION
**Pull request by SamirDroubi**
* When wrapping load/stores with guard, the code was doing a roundabout
    approach to get the block of the load/store which doesn't work when
    the load stage is simply a statement and not a loop nest. The correct
    way to get such a block is to simply use as_block()

### CODE REVIEW

**Comment by yamaguchi1024** [comment]
LGTM, but @skeqiqevian should take a look

File: tests/test_schedules.py [hunk 1]
```diff
@@ -2603,6 +2603,20 @@ def axpy(n: size, x: f32[n], y: f32[n]):
...
 
 
+def test_stage_mem_out_of_bound_point(golden):
+    @proc
+    def foo(n: size, m: size, x: f32[n], y: f32[n]):
+        assert m >= n
+        for i in seq(0, m):
+            if i < n:
+                y[i] = x[i]
+            if i < n:
```
Comment (skeqiqevian) [inline tests/test_schedules.py:2613]: Is there reason we need two `if i < n: y[i] = x[i]` stmts for this test?

Comment (SamirDroubi) [reply tests/test_schedules.py:2613]: That's a good point. I can't think of why I wrote it this way lol.

Comment (SamirDroubi) [reply tests/test_schedules.py]: Just fixed it.

**Pull request merged**